docs/source/daml-integration-kit/index.rst

```diff
@@ -0,0 +1,371 @@
+ledger's integrity and therefore need to validate DAML transactions.
+
+
+Message flow
+------------
+
+**TODO (BETA):**
+
+- explain to readers the life of a transaction at a high-level, so they have a
+  mental framework in place when looking at the example code.
+  (`GitHub issue <https://github.com/digital-asset/daml/issues/672>`__)
+
+
+Implementing your server and validator
+======================================
+
+Each `X` ledger requires at least the implementation of a specific
+`daml-on-<X>-server`. It might also require the implementation of a
+`<X>-daml-validator` as explained above. We provide two kinds of code to
+simplify their implementation.
+
+1. We provide Scala libraries for validating DAML transactions and serving the
+   Ledger API given implementations of two specific interfaces. See
+   :ref:`below <integration-kit_library_overview>` for an overview of these
+   libraries.
+
+2. We provide the `digital-asset/daml-on-x-example <https://github.com/digital-asset/daml-on-x-example>`__ GitHub repository,
+   which contains a complete example of a DAML Ledger backed by an in-memory
+   key-value store. It builds on our Scala libraries and demonstrates how they
+   can be assembled to serve the Ledger API and validate DAML transactions.
+
+   For ledgers where data is shared between all participant nodes, we
+   recommend using this example as a starting point for implementing your
+   server and validator. For ledgers with stronger privacy models, this example
+   can serve as an inspiration. You will need to dive deeper into how
+   transactions are represented and how to communicate them to
+   implement :doc:`DAML's privacy model </concepts/ledger-model/ledger-privacy>`
+   at the ledger level instead of just at the Ledger API level.
+
+Implementing your own server and validator requires a significant amount
+of context to be learned before you can decide on your architecture and
+implement it.
+
+
+Getting started
+---------------
+
+We recommend the following three steps to gain the knowledge necessary to
+implement your own server and validator.
+
+1. Complete the :doc:`/getting-started/quickstart`.
+2. Get an in-depth understanding of the :doc:`/concepts/ledger-model/index`.
+3. Build a mental model of how the :doc:`/app-dev/ledger-api-introduction/index`
+   is used to build DAML Applications.
+
+
+.. _integration-kit_writing_code:
+
+Deciding on the architecture and writing the code
+-------------------------------------------------
+
+Once you've completed the above steps, we recommend following these four steps
+to implement your own server and validator.
+
+1. Clone our example DAML Ledger backed by an in-memory key-value store from
+   the `digital-asset/daml-on-x-example <https://github.com/digital-asset/daml-on-x-example>`__.
+
+   **TODO (ALPHA):** create this example repository
+   (`issue <https://github.com/digital-asset/daml/issues/139>`__)
+
+2. Read the example code jointly with
```

Comment:
Given this, I think it makes sense to have these instructions _first_, then the architectural overview _later_. It could even be its own page.